tls/bignum.c

```diff
@@ -1228,12 +1228,8 @@ ttls_mpi_exp_mod(TlsMpi *X, const TlsMpi *A, const TlsMpi *E, const TlsMpi *N,
 	 * If 1st call, pre-compute R^2 mod N
 	 */
 	BUG_ON(!RR);
-	if (unlikely(ttls_mpi_empty(RR))) {
-		ttls_mpi_alloc(RR, N->used * 2 + 2);
-		ttls_mpi_lset(RR, 1);
-		ttls_mpi_shift_l(RR, RR, N->used * 2 * BIL);
-		ttls_mpi_mod_mpi(RR, RR, N);
-	}
+	if (unlikely(ttls_mpi_empty(RR)))
+		ttls_mpi_sq_mod(RR, N);
```

Comment:
This function is called via generic PK interface and we also call the function for DHM handshakes, so don't we have the same issue with DHM? Probably DHM fix is also required and we can just replace the initialization code with `BUG_ON(ttls_mpi_empty(RR))`.

Reply:
In out implementation DHM is only used for keys with `id-ecDH` oid, but openssl, boringSSl doesn't support this key types.

Reply:
Fixed in second commit, no race was here, because dhm context is always new for new connection, but fix other errors